hudi-utilities/src/main/java/org/apache/hudi/utilities/TableSizeStats.java

```diff
@@ -0,0 +1,411 @@
+      throw new HoodieIOException("Cannot read properties from dfs from file " + propsPath, ioe);
+    }
+    return filePaths;
+  }
+
+  private void logTableStats(String basePath, LocalDate startDate, LocalDate endDate, boolean partitionStats) throws IOException {
+
+    LOG.warn("Processing table " + basePath);
+    HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder()
+        .enable(false)
+        .build();
+    HoodieSparkEngineContext engineContext = new HoodieSparkEngineContext(jsc);
+    HoodieTableMetadata tableMetadata = HoodieTableMetadata.create(engineContext, metadataConfig, basePath,
+        FileSystemViewStorageConfig.SPILLABLE_DIR.defaultValue());
+    SerializableConfiguration serializableConfiguration = new SerializableConfiguration(jsc.hadoopConfiguration());
+
+    List<String> allPartitions = tableMetadata.getAllPartitionPaths();
+
+    final Histogram tableHistogram = new Histogram(new UniformReservoir(1_000_000));
+    allPartitions.forEach(partition -> {
+      // Partition name should conform to date format if startDate and/or endDate are specified. Otherwise, we don't
+      // need to parse partition name as date.
+      String dateString = partition;
+      if (partition.contains("=")) {
+        // Partition name may be in the form of "<column>=<date>". Try parsing out date.
+        String[] parts = partition.split("=");
+        dateString = parts[1].trim();
+      }
+
+      LocalDate partitionDate = null;
+      try {
+        if (startDate != null || endDate != null) {
+          partitionDate = LocalDate.parse(dateString, DATE_FORMATTER);
+        }
+        LOG.info("Parsed " + partition);
+      } catch (DateTimeParseException dtpe) {
+        LOG.error("Partition name {} must conform to date format if --start-date, --end-date, or --num-days are specified. ", partition, dtpe);
+      }
+
+      // Compute file size stats for all files in this partition if:
+      // 1. partition date is null (startDate and endDate are both null and not specified)
+      // 2. endDate is null (not specified) and partition date is equal to or after startDate.
+      // 3. startDate is null (not specified) and partition date is before endDate.
+      // 4. startDate and endDate are both specified and partition date lies in the range [startDate, endDate)
+      if (partitionDate == null
+          || (endDate == null && (partitionDate.isEqual(startDate) || partitionDate.isAfter(startDate)))
+          || (startDate == null && (partitionDate.isBefore(endDate)))
+          || (startDate != null && endDate != null && (partitionDate.isEqual(startDate) || partitionDate.isAfter(startDate) && partitionDate.isBefore(endDate)))) {
+        HoodieTableMetaClient metaClientLocal = HoodieTableMetaClient.builder()
+            .setBasePath(basePath)
+            .setConf(serializableConfiguration.get()).build();
+        HoodieMetadataConfig metadataConfig1 = HoodieMetadataConfig.newBuilder()
+            .enable(false)
```

Comment:
we should fetch this from table props and use it. 
```
boolean metadataEnabled = !this.getTableConfig().getMetadataPartitions().isEmpty() && this.getTableConfig().getMetadataPartitions().contains("files")
```

Reply:
Fixed.